airbyte-commons-worker/src/main/java/io/airbyte/workers/helper/UpdateConnectorConfigHelper.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.helper;
+
+import com.google.common.hash.Hashing;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.DestinationApi;
+import io.airbyte.api.client.generated.JobsApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.DestinationIdRequestBody;
+import io.airbyte.api.client.model.generated.DestinationRead;
+import io.airbyte.api.client.model.generated.DestinationUpdate;
+import io.airbyte.api.client.model.generated.JobIdRequestBody;
+import io.airbyte.api.client.model.generated.JobInfoLightRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.api.client.model.generated.SourceUpdate;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.Config;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for workers to persist updates to Source/Destination configs emitted from
+ * AirbyteControlMessages.
+ *
+ * This is in order to support connectors updating configs when running commands, which is specially
+ * useful for migrating configuration to a new version or for enabling connectors that require
+ * single-use or short-lived OAuth tokens.
+ */
+public class UpdateConnectorConfigHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(UpdateConnectorConfigHelper.class);
+
+  private final AirbyteApiClient apiClient;
+
+  public UpdateConnectorConfigHelper(final AirbyteApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  private UUID getConnectionIdFromJobId(final Long jobId) {
+    final JobsApi jobsApi = apiClient.getJobsApi();
+    final JobIdRequestBody body = new JobIdRequestBody().id(jobId);
+    final JobInfoLightRead jobInfo = AirbyteApiClient.retryWithJitter(
+        () -> jobsApi.getJobInfoLight(body),
+        "get job info");
+    return UUID.fromString(jobInfo.getJob().getConfigId());
+  }
+
+  /**
+   * Updates the Source from a sync job ID with the provided Configuration. Secrets and OAuth
+   * parameters will be masked when saving.
+   */
+  public void updateSource(final Long jobId, final Config config) {
+    final UUID connectionId = getConnectionIdFromJobId(jobId);
+
+    final ConnectionRead connection = AirbyteApiClient.retryWithJitter(
```

Comment:
Discussed offline - we will be blocking for now and revisit later since control messages should not be a common thing during the sync (_maybe_ 1 per hour depending on the connector)